### PR TITLE
Support platform for XCFrameworkInfoPlist

### LINKED ISSUE #11: XCFrameworkInfoPlist support more platforms
To resolve https://github.com/tuist/tuist/pull/6133, we need to be aware of the platform of an XCFramework. Tuist uses XcodeGraph to detect it.

However, the `XCFrameworkInfoPlist` currently defined the [platform](https://github.com/Binlogo/XcodeGraph/blob/614c4194dde1ef1df51c942b0eb2b343646ec213/Sources/XcodeGraph/Models/XCFrameworkInfoPlist.swift#L19), but this enum is complete, and not being used yet.

```swift
    /// It represents the library's platform.
    public enum Platform: String, Hashable, Codable {
        case ios
    }
```

### PR DESCRIPTION
**Pull request by Binlogo**
Resolve #11

Since the XCFramework is a bundle support for multiple platforms, it‘s difficult to say which binary is the primary binary. It is better to keep the platform in the binary library and defer which to use when it is needed.

### CODE REVIEW

**Review by fortmarek — APPROVED** [review]
Looks good. Let's merge this once we get approval for the related PR: https://github.com/tuist/tuist/pull/6414

**Review by pepicrft — APPROVED** [review]

**Comment by pepicrft** [comment]
I'm creating [a new release](https://github.com/tuist/XcodeGraph/pull/13) with the changes. Thanks a lot @Binlogo for these.

**Pull request merged**
